BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c

```diff
@@ -551,36 +549,18 @@ FindPartitions (
     return Status;
   }
 
-  // Read GPT Partition first
-  Buffer = (UINT8 *) PartBlockDev->BlockData;
-  Status = MediaReadBlocks (HwDevice, 1, DevBlockInfo.BlockSize, Buffer);
-  if (!EFI_ERROR (Status)) {
-    // Check GPT partition
-    Gpt = (EFI_PARTITION_TABLE_HEADER *)Buffer;
-    if ((UINT32)Gpt->Header.Signature == 0x20494645) {
-      Status = FindGptPartitions (PartBlockDev);
-    } else {
-      Status = FindMbrPartitions (PartBlockDev);
-    }
-
-    // Check result
-    if (EFI_ERROR (Status)) {
-      // Could not find any partition, so assume no partitions
-      BlockDev              = & (PartBlockDev->BlockDevice[0]);
-      BlockDev->StartBlock  = 0;
-      BlockDev->LastBlock   = DevBlockInfo.BlockNum - 1;
-      PartBlockDev->PartitionType    = EnumPartTypeUnknown;
-      PartBlockDev->PartitionChecked = TRUE;
-      PartBlockDev->BlockDeviceCount = 1;
-      Status = EFI_SUCCESS;
-    }
+  // Could not find any partition, so assume no partitions

```

Comment:
This is not correct.  Why did you remove all the partition check ?
It is still required for other devices, such as USB, SATA, UFS, etc.  It is a common function for all.  We just need to handle SPI/Mem device specially.  Remaining logic should be kept untouched.

Reply:
Ok. I considered only SPI/MEM. I will revert this change because if SPI/MEM it is handled earlier.